task/sql-tasks.js

```diff
@@ -280,7 +464,26 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT DISTINCT
+            a.CompanyName,
+            Products.ProductName AS 'ProductName',
+            OrderDetails.UnitPrice AS 'PricePerItem'
+        FROM Customers a
+        INNER JOIN Orders ON Orders.CustomerID = a.CustomerID
+        INNER JOIN OrderDetails ON OrderDetails.OrderID = Orders.OrderID
+        INNER JOIN Products ON Products.ProductID = OrderDetails.ProductID
+        WHERE OrderDetails.UnitPrice = (
+            SELECT 
+                MAX(OrderDetails.UnitPrice) 
+            FROM Customers b
+            INNER JOIN Orders ON Orders.CustomerID = b.CustomerID
+            INNER JOIN OrderDetails ON OrderDetails.OrderID = Orders.OrderID
+            WHERE b.CompanyName = a.CompanyName

```

Comment:
company name doesn't uniquely identify customer. also this column doesn't have an index so it's also slow.